### PR TITLE
add cypress e2e test for 526 which covers revisedAddDisabilities page

### PR DESCRIPTION
**Pull request by lukey-luke**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- Runs an automated cypress e2e test that will turn on feature flag `disability_526_improved_autosuggestions_add_disabilities_page` for [revisedAddDisabilities](https://github.com/department-of-veterans-affairs/vets-website/blob/cc-revisedAddDisabilities-cypress-testing/src/applications/disability-benefits/all-claims/pages/addDisabilitiesRevised.js#L32) and completes a 526-ez for a new condition.
- This will load the new version of the page that uses [Combobox.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/cc-revisedAddDisabilities-cypress-testing/src/applications/disability-benefits/all-claims/components/ComboBox.jsx#L27) for autosuggestions.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/496

## Testing done

- open cypress console via `yarn cy:open`
- navigate to `addDisabilitiesRevised.unit.spec.jsx` and double click
- observe cypress browser

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._
Screenshot 2024-06-03 at 4 05 02 PM
Screenshot 2024-06-03 at 4 16 33 PM

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by christinec-fftc** [comment]
Couple questions on this pr if you have time tomorrow

File: src/applications/disability-benefits/all-claims/tests/all-claims-revised-add-disabilities.cypress.spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    dataSets: ['newOnly-test'],
+
+    fixtures: {
+      data: path.join(__dirname, 'fixtures', 'data'),
+    },
+
+    pageHooks: pageHooks(cy),
+    setupPerTest: () => {
+      cy.login(mockUser);
+      setup(cy, mockFeatureToggles);
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/all-claims-revised-add-disabilities.cypress.spec.js:33]: hmm, i think on lines 30 and 33 we want to pass in `toggles`

Comment (lukey-luke) [reply src/applications/disability-benefits/all-claims/tests/all-claims-revised-add-disabilities.cypress.spec.js:33]: 👍

**Review by tjheffner — DISMISSED** [review]
LGTM

File: src/applications/disability-benefits/all-claims/tests/all-claims-revised-add-disabilities.cypress.spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    dataPrefix: 'data',
+    useWebComponentFields: true,
+
+    dataSets: ['newOnly-test'],
+
+    fixtures: {
+      data: path.join(__dirname, 'fixtures', 'data'),
+    },
+
+    pageHooks: pageHooks(cy, mockFeatureToggles),
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/all-claims-revised-add-disabilities.cypress.spec.js:30]:
```suggestion
    pageHooks: pageHooks(cy, toggles),
```

File: src/applications/disability-benefits/all-claims/tests/all-claims-revised-add-disabilities.cypress.spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    setupPerTest: () => {
+      cy.login(mockUser);
+      setup(cy, mockFeatureToggles);
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/all-claims-revised-add-disabilities.cypress.spec.js:33]:
```suggestion
      setup(cy, toggles);
```

The test runs successfully, but we need these changes to enable the revised page

**Review by christinec-fftc — APPROVED** [review]

**Pull request merged**